src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskManager.scala

```diff
@@ -41,8 +42,7 @@ class TaskManager @Inject()(val listeningExecutor: ListeningScheduledExecutorSer
 
   val taskCache = CacheBuilder.newBuilder().maximumSize(5000L).build[String, TaskState]()
 
-  val taskMapping = new mutable.HashMap[String, mutable.ListBuffer[(String, Future[_])]] with
-    collection.mutable.SynchronizedMap[String, mutable.ListBuffer[(String, Future[_])]]
+  val taskMapping: concurrent.Map[String, mutable.ListBuffer[(String, Future[_])]] = new ConcurrentHashMap().asScala
```

Comment:
This is true in enqueue, because the synchronization is on queues and not on taskMapping. Thx for double checking this.